### PR TITLE
test(exex): some small enhancements for finished height

### PR DESCRIPTION
**Pull request by tcoratger**
- Default for `ExExEvent` is useful especially for testing scenarios.
- `NoExExs` in `FinishedExExHeight` should be defined as default setting for simplicity as a no brainer.
- `From<u64> for FinishedExExHeight` is really practical to manipulate stuff in an easy manner.

### CODE REVIEW

File: crates/exex/exex/src/event.rs [hunk 1]
```diff
@@ -11,3 +11,9 @@ pub enum ExExEvent {
     /// On reorgs, it's possible for the height to go down.
     FinishedHeight(BlockNumber),
 }
+
+impl Default for ExExEvent {
```
Comment (shekhirin) [inline crates/exex/exex/src/event.rs:15]: i'm fine with it, but please implement only for `cfg(test)`

File: crates/exex/types/src/finished_height.rs [hunk 2]
```diff
@@ -1,9 +1,10 @@
 use alloy_primitives::BlockNumber;
 
 /// The finished height of all `ExEx`'s.
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, Default)]
 pub enum FinishedExExHeight {
     /// No `ExEx`'s are installed, so there is no finished height.
+    #[default]
```
Comment (shekhirin) [inline crates/exex/types/src/finished_height.rs:7]: I don't think we should rely on Default in this case, I would prefer to have an explicit enum variant in every scenario

**Review by shekhirin — CHANGES_REQUESTED** [review]

File: crates/exex/exex/src/event.rs (hunk 1 above)
Comment (tcoratger) [reply crates/exex/exex/src/event.rs:16]: Done

File: crates/exex/types/src/finished_height.rs (hunk 2 above)
Comment (tcoratger) [reply crates/exex/types/src/finished_height.rs]: Understood, just reverted

File: crates/exex/types/src/finished_height.rs [hunk 3]
```diff
@@ -23,3 +23,9 @@ impl FinishedExExHeight {
         matches!(self, Self::NotReady)
     }
 }
+
+impl From<u64> for FinishedExExHeight {
+    fn from(value: u64) -> Self {
+        Self::Height(value)
+    }
+}
```
Comment (onbjerg) [inline crates/exex/types/src/finished_height.rs:31]: i'm not entirely convinced this is a good idea, we might add more variants that also just take a u64 so this is a potential future footgun. i get that this is a good idea right now, but i'd rather not have to break the api so "severely" in the future if possible

**Review by onbjerg — CHANGES_REQUESTED** [review]

File: crates/exex/types/src/finished_height.rs (hunk 3 above)
Comment (shekhirin) [reply crates/exex/types/src/finished_height.rs:31]: right, good point. Can this PR be closed then?

Comment (tcoratger) [reply crates/exex/types/src/finished_height.rs:31]: Feel free to close if this is a blocker, was just a small 5 min enhancement

**Pull request closed without merging**